src/python/pants/option/global_options.py

```diff
@@ -216,6 +216,11 @@ def register_bootstrap_options(cls, register):
              help='The path to the watchman UNIX socket. This can be overridden if the default '
                   'absolute path length exceeds the maximum allowed by the OS.')
 
+    # This option changes the parser behavior in a fundamental way (which currently invalidates
+    # all caches), and needs to be parsed out early, so we make it a bootstrap option.
+    register('--build-file-imports', choices=['allow', 'warn', 'error'], default='warn',
```

Comment:
Given that you've gone through the trouble to tokenize (thanks!), how would you feel about having two options here:
```
--build-file-tokens-warn=['while']
--build-file-tokens-error=['import']
```
?

If we think that's overkill, then a-ok.

Reply:
I'm going to save this for the future; it's just cosmetic to retrofit this, but I'd want a more thorough set of tests before exposing arbitrary tokens through a public interface.